### ISSUE TITLE
[Suggestion] Dictionary initialization

### ISSUE THREAD

**Issue by AndyPook** [confluentinc/confluent-kafka-dotnet#683]
Description
===========
A code pattern suggestion.
There are several places where Dictionary<,>'s are created. eg in Config
`public ClientConfig(ClientConfig config) { this.properties = new Dictionary<string, string>(config.ToDictionary(a => a.Key, a => a.Value)); }`

This pattern causes two dictionaries to be created and iterated over. `ToDictionary` already creates a new Dictionary. Wrapping that result in another Dictionary just iterates it and throws it away. So the following is equivalent.
`public ClientConfig(ClientConfig config) { this.properties = config.ToDictionary(a => a.Key, a => a.Value); }`
Or, preferably, as `.properties` is within scope
`public ClientConfig(ClientConfig config) { this.properties = new Dictionary<string, string>(config.properties); }`

Also, decendent classes, like `AdminClientConfig` should really have their ctor's call `base` rather than reimplementing. ie
`public AdminClientConfig(ClientConfig config) : base(config) { }`

As a result `.properties` can then be made `private`.

How to reproduce
================

`var x = new ClientConfig(myOtherConfig);`
will create multiple dictionaries while initializing the `.properties` member.

Checklist
=========
Please provide the following information:

 - [ ] Confluent.Kafka nuget version:
 - [ ] Apache Kafka version: 
 - [ ] Client configuration:
 - [ ] Operating system:
 - [ ] Provide logs (with "debug" : "..." as necessary in configuration)
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by AndyPook** [comment]
happy to provide PR if interested

**Comment by mhowlett** [comment]
good point, happy to accept a PR here too (but wait until #656 is in as it will clash). note that the config classes are only used on startup, so this is a pretty minor optimization.

**Comment by AndyPook** [comment]
agreed, low impact.
I have a changeset ready :) can rebase once #656 is done
